src/lib/qnio/nio_client.c

```diff
@@ -82,6 +83,16 @@ open_connection(struct network_channel *netch, int flags, int euid)
         goto out;
     }
 
+    if(cmn_ctx->ssl_ctx)
+    {
+        ssl = SSL_new(cmn_ctx->ssl_ctx);
+        SSL_set_fd(ssl, sock);
+        if (SSL_connect(ssl) == -1) {
+            nioDbg("ssl connect failed");
```

Comment:
It looks like is also leaked down in the out: error handling, and orphaned with the 'free(c)':

 out:
     if(sock != -1) {
         close(sock);
     }
     if(infos) {
         freeaddrinfo(infos);
     }
     if(c) {
         free(c);
     }